src/matchers/numbers.test.lua

```diff
@@ -0,0 +1,653 @@
+local matchers = require("src.matchers.numbers")
+local assertMatcher = require("src.asserts.matchers")
+
+local INFINITY = math.huge
+local NAN = 0 / 0
+
+describe("number matchers", function()
+	local CONTEXT = {
+		inverted = false,
+	}
+
+	local INVERTED_CONTEXT = {
+		inverted = true,
+	}
+
+	describe("toBeCloseTo", function()
+		local closeValue = 2.005
+		local testValue = 2
+
+		it(
+			"should pass when the received value is close to expected value",
+			function()
+				-- Given
+				local numDigits = 1
+
+				-- When
+				local result = matchers.toBeCloseTo(
+					CONTEXT,
+					closeValue,
+					testValue,
+					numDigits
+				)
+
+				-- Then
+				assertMatcher.passed(result)
+			end
+		)
+
+		describe("should fail", function()
+			test(
+				"when the received value is not close to the expected value",
+				function()
+					-- Given
+					local numDigits = 3
+
+					-- When
+					local result = matchers.toBeCloseTo(
+						CONTEXT,
+						closeValue,
+						testValue,
+						numDigits
+					)
+
+					-- Then
+					assertMatcher.failed(result)
+					assertMatcher.hasMessage(
+						result,
+						("Expected %.3f to be close to %d within %d digits"):format(
+							closeValue,
+							testValue,
+							numDigits
+						)
+					)
+				end
+			)
+
+			test("on infinity for any parameter", function()
+				-- Given
+				local normalNumber = 1
+
+				-- When
+				local result =
+					matchers.toBeCloseTo(CONTEXT, normalNumber, INFINITY, 1)
+
+				local resultSwapped =
+					matchers.toBeCloseTo(CONTEXT, INFINITY, normalNumber, 1)
+
+				-- Then
+				assertMatcher.failed(result)
+				assertMatcher.hasMessage(
+					result,
+					("Expected %d to be close to inf within 1 digits"):format(
+						normalNumber
+					)
+				)
+
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected inf to be close to %d within 1 digits"):format(
+						normalNumber
+					)
+				)
+			end)
+
+			test("on NaN for any parameter", function()
+				-- Given
+				local normalNumber = 1
+
+				-- When
+				local result =
+					matchers.toBeCloseTo(CONTEXT, normalNumber, NAN, 1)
+
+				local resultSwapped =
+					matchers.toBeCloseTo(CONTEXT, NAN, normalNumber, 1)
+
+				-- Then
+				assertMatcher.failed(result)
+				assertMatcher.hasMessage(
+					result,
+					("Expected %d to be close to %s within 1 digits"):format(
+						normalNumber,
+						tostring(NAN)
+					)
+				)
+				assertMatcher.failed(resultSwapped)
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected %s to be close to %d within 1 digits"):format(
+						tostring(NAN),
+						normalNumber
+					)
+				)
+			end)
+		end)
```

Comment:
We should also test when both parameters are infinity or `NaN`.

`NaN` should not be close to `NaN`

infinity should be close to infinity, and -infinity should be close to -infinity
infinity should not be close to -infinity
Source: https://github.com/facebook/jest/pull/7444